midiproc/buffer.c

```diff
@@ -0,0 +1,253 @@
+//
+// Copyright(C) 2017 Alex Mayfield
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//     A simple buffer and reader implementation.
```

Comment:
The null-terminated string is the filename. I believe Windows has a maximum length on filenames, so you could just use that. (Other systems don't have such a limit, but we only care about Windows for this feature).

But: is the "write to temp file and send the path to play" approach the best one? Modern versions of SDL_mixer let you supply a rwops handle instead of a filename, allowing the temp file to be completely eliminated. Maybe you could just transfer the whole file contents over the pipe and then you don't need to send a filename anyway?